crates/app/src/main.rs

```diff
@@ -21,7 +22,8 @@ async fn main() -> anyhow::Result<()> {
         .get_matches();
     let config_path = matches.get_one::<String>("config").unwrap();
 
-    let settings = Settings::from(config_path).unwrap();
+    let fs = RealFileSystem;
+    let settings = Settings::from(&fs, config_path).unwrap();
```

Comment:
`Real`FileSystem seems to imply there exists a fake one, which is the information that doesn't need to be exposed to the users. What about making `File` object instead of FileSystem, and pass it to the `Settings::from`, like
```
Settings::from(LocalFile::new(config_path))
```

There could be another `InMemoryFile` which implements the same traits that can be used for testing.

Reply:
Refactored the file system to introduce `LocalFileSystem` and `InMemoryFileSystem`, and refactored `Settings::from` to take a `FileSource`.